keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,644 @@
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+This feature is not dependent on API server or etcd availability.
+
+###### What are other known failure modes?
+
+N/A
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+### Alternative 1: Add admission plugin that blocks gitRepo volumes
+We add an in-tree admission plugin that, when enabled, will prevent Pods with
+gitRepo volumes from being admitted.
+
+Pros:
+- Backwards compatible.
+- Allows cloud providers to make a call if they want to support gitRepo volumes or not.
+
+Cons:
+- Not secure by default, requires cluster-admin action.
+- The only escape valve is cluster-scoped.
+- We'll be adding an in-tree admission plugin for a volume type that is not
+  maintained.
+
+### Alternative 2: Use ValidatingAdmissionPolicy
+We publish a VAP which prevents Pods and other workloads from using gitRepo, and
+strongly recommend cluster-admins (or cluster-providers) install it.
+
+Pros:
+- Backwards compatible.
+- Allows cluster-admins to make a call if they want to support gitRepo volumes or not.
+- Can be scoped to namespaces rather than whole clusters.
+- Works on older clusters.
+
+Cons:
+- Not secure by default
+- Requires cluster-admin action.
+
+### Alternative 3: Make Git hooks disableable
+We add a new field to gitRepo volume called `enableHooks` to
+configure if Git hooks should be run as part of gitRepo volume mounting.
+When this value is set to true kubelet will execute Git hooks when performing 
+`git clone` and `git checkout`. If enableHooks is unset or set to false, kubelet
+will not run any Git hooks when performing `git clone` and `git checkout`.
+
+Pros:
+- gitRepo volume type will still be supported and we will eliminate the risks.
+
+Cons:
```

Comment:
I would also add something along the lines of: "git hooks may not be the only source of vulnerability in the future and we will be playing whack-a-mole with executing git as kubelet/host-root"

Reply:
Done.